sound/soc/sof/intel/hda-dai-ops.c

```diff
@@ -237,6 +237,36 @@ static unsigned int generic_calc_stream_format(struct snd_sof_dev *sdev,
 	return format_val;
 }
 
+static unsigned int dmic_calc_stream_format(struct snd_sof_dev *sdev,
+					    struct snd_pcm_substream *substream,
+					    struct snd_pcm_hw_params *params)
+{
+	unsigned int format_val;
+	unsigned int channels;
+	unsigned int format;
+	unsigned int width;
+
+	channels = params_channels(params);
+	format = params_format(params);
```

Comment:
Sparse issue @juimonen 

sound/soc/sof/intel/hda-dai-ops.c:250:16: error: incorrect type in assignment (different base types)

I think this should be defined with snd_pcm_format_t  or use a different macro